src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -209,7 +150,7 @@ internal bool IsKeyValuePairAdapter
         internal bool CreateNewInstanceViaDefaultConstructor([NotNullWhen(true)] out object? obj)
         {
             ConstructorInfo? ci = GetNonAttributedTypeConstructor();
-            if (ci == null)
+            if (ci == null || UnderlyingType == Globals.TypeOfSchemaDefinedType)
             {
```

Comment:
This is new code in Core, how did you determine the additional clause was needed here?

Reply:
This was my offhanded attempt at guarding against using the dummy type that got introduced for creating contracts when importing schemas. As mentioned in the PR description, this whole dummy-type pattern is not something I have thoroughly examined/reviewed yet.